src/test/client.js

```diff
@@ -330,6 +383,14 @@ function simulateConnection(dumpPath, resultPath, ua, protocolV) {
     });
 
     connection.connect();
+    connection.on('disconnect', () => {
+        this.disconnected = true;
+
+        // we need to wait a little bit before reconnecting.
+        setTimeout(() => {
+            connection.connect();
```

Comment:
There is one problem with how reconnect is currently handled, if the client reconnects after the server decided it waited enough then the FeatureExtractor will process the resulting dump file as if it was a new session (even though identity information is missing) resulting in an overwritten s3 dump with partial information and a duplicate entry in redshift. This can also happen if the server restarts and the client lands on another server because of how haproxy is currently setup. Ideally the server or the client would identify these cases and not process them, simply logging an error would do.
Is this case handled by client/server protocol somehow?